src/main/scala/org/scanet/optimizers/Algorithm.scala

```diff
@@ -1,10 +1,11 @@
 package org.scanet.optimizers
 
-import org.scanet.core.{Output, TensorType}
+import org.scanet.core.{Output, Tensor, TensorType}
 import org.scanet.math.Numeric
 
-trait Algorithm {
+trait Algorithm[S] {
```

Comment:
I've looked at other optimizers and thought that sometimes we would need pair a pair of tensors as "state". If this is not the case - it would be nice to remove type param

Reply:
makes sense